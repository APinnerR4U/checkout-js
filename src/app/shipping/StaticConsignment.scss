@import '../../../node_modules/@bigcommerce/citadel/src/settings/global/global';

.staticConsignmentContainer {
    ul {
        list-style: none;
        margin: 0;
    }

    & + .staticConsignmentContainer {
        border-top: container("border");
        margin-top: spacing("single");
        padding-top: spacing("single");
    }
}

.staticConsignment-items {
    margin: spacing("quarter") 0;
}
