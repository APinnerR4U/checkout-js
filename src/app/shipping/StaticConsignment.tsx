import { Cart, Consignment } from '@bigcommerce/checkout-sdk';
import React from 'react';

import { StaticAddress } from '../address';
import { TranslatedString } from '../language';

import StaticShippingOption from './shippingOption/StaticShippingOption';
import findLineItems from './util/findLineItems';
import getLineItemsCount from './util/getLineItemsCount';
import './StaticConsignment.scss';

interface StaticConsignmentProps {
    consignment: Consignment;
    cart: Cart;
    compactView?: boolean;
    showShippingMethod?: boolean;
}

const StaticConsignment: React.SFC<StaticConsignmentProps> = ({
    consignment,
    cart,
    showShippingMethod,
    compactView,
}) => {
    const lineItems = findLineItems(cart, consignment);
    const {
        shippingAddress: address,
        selectedShippingOption,
    } = consignment;

    return (
        <div className="staticConsignment">
            { !compactView &&
                <strong>
                    <TranslatedString id="shipping.shipping_address_heading" />
                </strong>
            }
            <StaticAddress address={ address } />
            { !compactView &&
                <div className="staticConsignment-items">
                    <strong>
                        <TranslatedString id="cart.item_count_text"
                            data={ { count: getLineItemsCount(lineItems) } } />
                    </strong>
                    <ul>
                        { lineItems.map(item =>
                            <li key={ item.id }>
                                { item.quantity } x { item.name }
                            </li>
                        ) }
                    </ul>
                </div>
            }

            { showShippingMethod && selectedShippingOption &&
                <div>
                    { !compactView &&
                        <strong>
                            <TranslatedString id="shipping.shipping_method_label"/>
                        </strong>
                    }
                    <div className="shippingOption shippingOption--alt">
                        <StaticShippingOption method={ selectedShippingOption } />
                    </div>
                </div>
            }
        </div>
    );
};

export default StaticConsignment;
