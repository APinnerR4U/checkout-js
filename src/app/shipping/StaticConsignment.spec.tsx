import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

import { getCart } from '../cart/carts.mock';

import { getConsignment } from './consignment.mock';
import StaticConsignment from './StaticConsignment';

describe('StaticConsignment Component', () => {
    const consignment = getConsignment();
    const cart = getCart();

    it('renders static consignment without shipping method', () => {
        const tree = shallow(
            <StaticConsignment cart={ cart } consignment={ consignment }/>);

        expect(toJson(tree)).toMatchSnapshot();
    });

    it('renders static consignment with shipping method', () => {
        const tree = shallow(
            <StaticConsignment cart={ cart } consignment={ consignment } showShippingMethod={ true }/>);

        expect(toJson(tree)).toMatchSnapshot();
    });

    it('renders compact view of static consignment', () => {
        const tree = shallow(
            <StaticConsignment cart={ cart } compactView={ true } consignment={ consignment } showShippingMethod={ true }/>);

        expect(toJson(tree)).toMatchSnapshot();
    });
});
