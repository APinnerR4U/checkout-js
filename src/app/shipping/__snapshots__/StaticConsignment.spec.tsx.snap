// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StaticConsignment Component renders compact view of static consignment 1`] = `
<div
  className="staticConsignment"
>
  <WithCheckout(StaticAddress)
    address={
      Object {
        "address1": "12345 Testing Way",
        "address2": "",
        "city": "Some City",
        "company": "Bigcommerce",
        "country": "United States",
        "countryCode": "US",
        "customFields": Array [],
        "firstName": "Test",
        "lastName": "Tester",
        "phone": "555-555-5555",
        "postalCode": "95555",
        "stateOrProvince": "California",
        "stateOrProvinceCode": "CA",
      }
    }
  />
  <div>
    <div
      className="shippingOption shippingOption--alt"
    >
      <StaticShippingOption
        method={
          Object {
            "cost": 0,
            "description": "Flat Rate",
            "id": "0:61d4bb52f746477e1d4fb411221318c3",
            "imageUrl": "",
            "isRecommended": true,
            "transitTime": "",
            "type": "shipping_flatrate",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`StaticConsignment Component renders static consignment with shipping method 1`] = `
<div
  className="staticConsignment"
>
  <strong>
    <WithLanguage(TranslatedString)
      id="shipping.shipping_address_heading"
    />
  </strong>
  <WithCheckout(StaticAddress)
    address={
      Object {
        "address1": "12345 Testing Way",
        "address2": "",
        "city": "Some City",
        "company": "Bigcommerce",
        "country": "United States",
        "countryCode": "US",
        "customFields": Array [],
        "firstName": "Test",
        "lastName": "Tester",
        "phone": "555-555-5555",
        "postalCode": "95555",
        "stateOrProvince": "California",
        "stateOrProvinceCode": "CA",
      }
    }
  />
  <div
    className="staticConsignment-items"
  >
    <strong>
      <WithLanguage(TranslatedString)
        data={
          Object {
            "count": 0,
          }
        }
        id="cart.item_count_text"
      />
    </strong>
    <ul />
  </div>
  <div>
    <strong>
      <WithLanguage(TranslatedString)
        id="shipping.shipping_method_label"
      />
    </strong>
    <div
      className="shippingOption shippingOption--alt"
    >
      <StaticShippingOption
        method={
          Object {
            "cost": 0,
            "description": "Flat Rate",
            "id": "0:61d4bb52f746477e1d4fb411221318c3",
            "imageUrl": "",
            "isRecommended": true,
            "transitTime": "",
            "type": "shipping_flatrate",
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`StaticConsignment Component renders static consignment without shipping method 1`] = `
<div
  className="staticConsignment"
>
  <strong>
    <WithLanguage(TranslatedString)
      id="shipping.shipping_address_heading"
    />
  </strong>
  <WithCheckout(StaticAddress)
    address={
      Object {
        "address1": "12345 Testing Way",
        "address2": "",
        "city": "Some City",
        "company": "Bigcommerce",
        "country": "United States",
        "countryCode": "US",
        "customFields": Array [],
        "firstName": "Test",
        "lastName": "Tester",
        "phone": "555-555-5555",
        "postalCode": "95555",
        "stateOrProvince": "California",
        "stateOrProvinceCode": "CA",
      }
    }
  />
  <div
    className="staticConsignment-items"
  >
    <strong>
      <WithLanguage(TranslatedString)
        data={
          Object {
            "count": 0,
          }
        }
        id="cart.item_count_text"
      />
    </strong>
    <ul />
  </div>
</div>
`;
